pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
> Is that only an issue with repeated create/deletes of a pod with an identical name?

No, it can happen to a single delete. At first I thought this would be rare enough, but it was reported several times.